library/Rules/Domain.php

```diff
@@ -27,18 +27,7 @@ public function __construct($tldCheck = true)
         $this->tldCheck($tldCheck);
         $this->otherParts = new AllOf(
             new Alnum('-'),
-            new Not(new StartsWith('-')),
-            new AnyOf(
-                new Not(
-                    new Contains('--')
-                ),
-                new AllOf(
-                    new StartsWith('xn--'),
-                    new Callback(function ($str) {
-                        return mb_substr_count($str, '--') == 1;
-                    })
-                )
-            )
+            new Not(new StartsWith('-'))
```

Comment:
You should also add `new Not(new EndsWith('-'))`, because `domain-.com` is also not valid